src/main/java/org/opensearch/neuralsearch/query/visitor/NeuralSearchQueryVisitor.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.query.visitor;
+
+import java.util.Map;
+
+import lombok.AllArgsConstructor;
+
+import org.apache.lucene.search.BooleanClause;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.index.query.QueryBuilderVisitor;
+import org.opensearch.neuralsearch.query.NeuralQueryBuilder;
+
+/**
+ * Neural Search Query Visitor. It visits the each and every component of query buikder tree.
```

Comment:
nit: "It visits the each and every component of query buikder tree." -> "It visits each and every component of query builder tree."